spec/features/merchants/items/index_spec.rb

```diff
@@ -58,4 +58,26 @@
     expect(page).to have_link("#{@item_2.name}", href: merchant_item_path(@merchant_1.id, @item_2.id))
     expect(page).to have_link("#{@item_3.name}", href: merchant_item_path(@merchant_1.id, @item_3.id))
   end
+
+  describe 'item disable/enable button' do 
+    scenario 'visitor sees a button to disable that item' do 
+      within "#enabled_item-#{@item_1.name}" do
+        expect(page).to have_button("Disable")
+        click_button "Disable"
+      end
+      
```

Comment:
Nice use of within blocks, and good job checking the result afterwards. 

I wonder if there is some piece of displayed information we can check for that changes from enabled to disable and back (ie check the order of items displayed on the page knowing that one moved further down the page, or check that before a button click a specific item has a displayed status of enable and after it has a displayed status of disable, or something like that).

Any thoughts or comments?

Reply:
There’s a Capybara Scoped Finder  | checkout this link: (https://gist.github.com/tomas-stefano/6652111)) | within that looks like it could probably make that happen. I’m not sure if I’d have to convert my form to a form_with model or not. Great Q!!!